src/util/htmlToJson.ts

```diff
@@ -120,9 +120,9 @@ const handleBr = (nodes: HimalayaNode[], brIndex: number): HimalayaNode[] => {
 }
 
 /** Append children to parent as children property if it's necessary. */
-const joinChildren = (nodes: Block[]) => {
+const joinChildren = (nodes: Block[] | Block[][]) => {
   // split by chunk with size of 2, first element in chunk is Block - parent, the second is Block[] - children
-  const chunks = _.chunk(nodes, 2)
+  const chunks = _.chunk(nodes as Block[], 2)
```

Comment:
You've just moved the invalid cast from `himilayaToBlock` to `joinChildren`. If the actual runtime type if `Block[] | [][]`, then you can't simply cast it to `Block[]`. You have to create a case if it's `Block[]`, and a case if it's `Block[][]`. (This also assumes that `Block[][][]` is not possible, which would require a recursive type.)

Reply:
> > I don't think it should be because it's main goal is to just append children to parent as children property. Also the blocks that are passed to this function has been recurssively calculated to get intended structure
> 
> Sure, but that only works if `joinChildren` is actually getting `Block[]`. If it is getting nested lists, it either needs to handle them or we need to convert to `Block[]` higher up.

That make sense. i will see what I can do here

Reply:
@karunkop When we paste the offending HTML, the type of `nodes` at runtime is `[Block, Block[], Block[]]`. You can confirm this yourself with a `console.log`. Since this does not match `Block[] | Block[][]`, it tells us that the type annotations are incorrect.

I would recommend tracing the value of `nodes` upstream until you find the first point where the runtime type does not match the type annotation. Then fix the type annotation there, and update the downstream types accordingly.